### ISSUE TITLE
Training loss = nan

### ISSUE THREAD

**Issue by zhiyi-set** [yaodongyu/TRADES#30]
Thank you for providing the codes, they are so excellent. But I met a problem, when I run _train_trades_cifar10.py_ using resnet, the training loss is always nan.

What I modified is the epsilon size, I used epsilon = 2/255, 4/255, 8/255, and 16/255. Do you know the reason? Thank you again

![image](https://user-images.githubusercontent.com/105244271/215294254-c253f683-789d-4498-a205-593d7e9c0b67.png)

**Comment by Zaid-Hameed** [comment]
Hi Zhiyi, 

Did you find the cause behind this error or any solution?

Thanks!

**Comment by zhiyi-set** [comment]
@Zaid-Hameed No, I try different seeds/initialization to train the model, and I got numerical loss only once or twice, most of the time it showed the nan loss. I didn't figure out the reasons, but you can try this way.

**Comment by Zaid-Hameed** [comment]
@Zhiyi-Dong-6  In my experiments, this issue occurs for pytorch releases >= 1.13 and is most likely happening because of modification in KLDivLoss implementation in pytorch. A very simple work around by either adding a small epsilon (~1e-8) or clamping minimum values by this epsilon in "trades.py" like this 

```
loss_robust = (1.0 / batch_size) * criterion_kl(F.log_softmax(model(x_adv), dim=1),
                                                    torch.clamp(F.softmax(model(x_natural), dim=1), min=EPS))

```                                                    
solves the issue. For details on this issue in pytorch please checkout this link: 
https://github.com/pytorch/pytorch/issues/89558

Hope it helps!

**Comment by zhiyi-set** [comment]
@Zaid-Hameed Thanks!

**Comment by youyuanyi** [comment]
> @Zhiyi-Dong-6 In my experiments, this issue occurs for pytorch releases >= 1.13 and is most likely happening because of modification in KLDivLoss implementation in pytorch. A very simple work around by either adding a small epsilon (~1e-8) or clamping minimum values by this epsilon in "trades.py" like this
> 
> ```
> loss_robust = (1.0 / batch_size) * criterion_kl(F.log_softmax(model(x_adv), dim=1),
>                                                     torch.clamp(F.softmax(model(x_natural), dim=1), min=EPS))
> ```
> 
> solves the issue. For details on this issue in pytorch please checkout this link: [pytorch/pytorch#89558](https://github.com/pytorch/pytorch/issues/89558)
> 
> Hope it helps!

hello, when I traing a ResNet34, the loss is nan. I found this issue and woner about is this  'EPS' value of 8/255 or 1e-8

**Comment by youyuanyi** [comment]
> @Zhiyi-Dong-6 In my experiments, this issue occurs for pytorch releases >= 1.13 and is most likely happening because of modification in KLDivLoss implementation in pytorch. A very simple work around by either adding a small epsilon (~1e-8) or clamping minimum values by this epsilon in "trades.py" like this
> 
> ```
> loss_robust = (1.0 / batch_size) * criterion_kl(F.log_softmax(model(x_adv), dim=1),
>                                                     torch.clamp(F.softmax(model(x_natural), dim=1), min=EPS))
> ```
> 
> solves the issue. For details on this issue in pytorch please checkout this link: [pytorch/pytorch#89558](https://github.com/pytorch/pytorch/issues/89558)
> 
> Hope it helps!

Hello, when I train a ResNet34, the loss is nan. I found this issue and wanted to know if this "EPS" value was 8/255 or 1e-8
